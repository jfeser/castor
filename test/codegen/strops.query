select([strlen("test") as l, strpos("testing", "in") as p], ascalar(0 as z))