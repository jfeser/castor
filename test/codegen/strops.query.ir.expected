// Locals:
// hoisted0 : int (persists=false)
// hoisted1 : int (persists=false)
// hoisted2 : tuple[int, int] (persists=false)
fun printer () : void {
    hoisted0 = strlen("test");
    hoisted1 = strpos("testing", "in");
    hoisted2 = (hoisted0, hoisted1);
    print(tuple[int, int], hoisted2);
}
// Locals:
// hoisted3 : int (persists=false)
// hoisted4 : int (persists=false)
// hoisted5 : tuple[int, int] (persists=false)
fun consumer () : void {
    hoisted3 = strlen("test");
    hoisted4 = strpos("testing", "in");
    hoisted5 = (hoisted3, hoisted4);
    consume(tuple[int, int], hoisted5);
}
